Formula/lib/libtensorflow.rb

```diff
@@ -73,29 +78,23 @@ def install
       ]
     end
     targets = %w[
-      tensorflow:libtensorflow.so
-      tensorflow:install_headers
-      tensorflow/tools/benchmark:benchmark_model
-      tensorflow/tools/graph_transforms:summarize_graph
-      tensorflow/tools/graph_transforms:transform_graph
+      //tensorflow/tools/lib_package:libtensorflow
+      //tensorflow/tools/benchmark:benchmark_model
```

Comment:
The `//tensorflow/tools/lib_package:libtensorflow` is the target used to generate `libtensorflow` (C API) release:
- https://www.tensorflow.org/install/lang_c
- https://github.com/tensorflow/tensorflow/tree/master/tensorflow/tools/lib_package
- https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/ci_build/builds/libtensorflow.sh#L76

This does remove various extra headers that were previously installed but most were not usable (e.g. C++ headers when we don't build C++ libs).

Aligns to previous discussion in https://github.com/Homebrew/homebrew-core/pull/82526